acme/acme/dns_resolver_test.py

```diff
@@ -21,33 +25,56 @@ def create_txt_response(name, txt_records):
     return dns.rrset.from_text_list(name, 60, "IN", "TXT", txt_records)
 
 
-class TxtRecordsForNameTest(unittest.TestCase):
+@test_util.skip_unless(DNS_AVAILABLE,
+                       "optional dependency dnspython is not available")
+class DnsResolverTestWithDns(unittest.TestCase):
+    """Tests for acme.dns_resolver when dns is available."""
+    @classmethod
+    def _call(cls, name):
+        from acme import dns_resolver
```

Comment:
Running the function we're testing like this is a fairly common thing in this project. The basic idea is to import whatever you're testing in the smallest scope possible to minimize other actions having side effects on what you're testing.

I don't think putting it in `setUp` wouldn't work unless we did something like:
```
def setUp(self):
    from acme import dns_resolver
    self.txt_records_for_name = dns_resolver.txt_records_for_name
```
And then called the function through `self` for all the tests.

We could potentially try and import the name as a class attribute, but I went with `_call` as it's more common in the code base and has the minor benefit I mentioned above.